problems/[230126] 32. Longest Valid Parentheses.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * leetcode problem link: undefined
+ *
+ * @param {string} s
+ * @return {number}
+ */
+const longestValidParentheses = function(s) {
+  if (s.length <= 1) return 0;
+
```

Comment:
```suggestion
var longestValidParentheses = function(s) {
  const stack = [-1];
  let maxLength = 0;

  for (let i = 0; i < s.length; i++) {
    if (s[i] === "(") {
      stack.push(i);
    } else {
      stack.pop();

      if (stack.length) {
        const length = i - stack[stack.length - 1];
        maxLength = Math.max(maxLength, length);
      } else {
        stack.push(i);
      }
    }
  }
  
  return maxLength;
};
```

두 번째 방법은 저도 이해는 했는데 이걸 내 머릿속으로 생각해내진 못하겠다.. 라는 방법이었는데

첨에 stack 초기값으로 -1을 넣어줘요
루프를 돌면서 왼쪽 괄호면 stack에 "index"를 넣어줍니다
오른쪽 괄호일 때 stack pop을 해줍니다.
1. stack에 아이템이 있으면 
- 현재 index - stack 맨 끝에 들어있는 값과 count 중 큰 값을 count로 설정해줘요.
2. stack에 아이템이 없으면 현재 index를 stack에 push해요
반복..
((())) 의 괄호라고 했을 때

**루프**
0번째 index = "(", stack = [-1, 0])
1번째 index = "(", stack = [-1, 0, 1])
2번째 index = "(", stack = [-1, 0, 1, 2])
3번째 index = ")" 
-> pop -> stack = [-1, 0, 1]
-> count = 2 (3 - 1)
4번째 index = ")" 
-> pop -> stack = [-1, 0]
-> count = 4 (4 - 0)
5번째 Index = ")"
-> pop -> stack = [-1]
-> count = 6 (5 - -1)

정답 6 리턴

이 방법은 어케 하는진 알겠는데 사실 왜 이런 방법을 생각해냈는지 이해하지 못했어요.. ( ఠ‿ఠ )

Reply:
인덱스를 스택에 넣다니 이건 정말 신박하군요..!
새삼 다양한 풀이법이 존재한다는게 재밌네요 >< 다음 문제는 우리 둘 다 한번 해결해봐요!!!
오늘도 수고 많으셨습니다 현정님 다음주 바코 출입은 불가능하겠죠..? 삼성역 놀러갈게요 우리 곧 만나용💕